serverv2/transaction/valdiation.go

```diff
@@ -0,0 +1,44 @@
+package transaction
+
+import (
+	"context"
+
+	tx "cosmossdk.io/core/transaction"
+)
+
+var _ tx.Validator[tx.Tx] = TxValidator[tx.Tx]{}
+
+// TxValidator is a transaction validator that validates transactions based off an existing set of handlers
+// TxValidator is designed to be synchronous
+type TxValidator[T tx.Tx] struct {
+	handler tx.Handler
+}
+
+func NewTxValidator[T tx.Tx]() *TxValidator[T] {
+	return &TxValidator[T]{}
+}
+
+// RegisterHandler registers the handlers to the transaction verifier.
+// The order of the handlers is important. The order passed here is the order of execution
+func (v *TxValidator[T]) RegisterHandler(h tx.Handler) {
+	v.handler = h
+}
+
+// Validate validates the transaction
+// it returns the context to be used further for execution
+func (v TxValidator[T]) Validate(ctx context.Context, txs []T, simulate bool) (context.Context, map[[32]byte]error) {
+	var (
+		errMap = make(map[[32]byte]error, len(txs)) // used to return a map of which txs failed
+		uctx   = ctx                                // retrun the context to be used further for execution
+	)
+
+	for _, tx := range txs {
```

Comment:
@tac0turtle why not consider just using `errors.Join`?

```go
for _, tx := range txs {
   ctx, txErr := v.handler(ctx, tx, simulate)
   err = errors.Join(err, txErr)
   newCtx = ctx
}
```

Reply:
it would need an error that is specific to the single tx. I dont think there is a clean way to say tx5,tx6,tx7 failed out of a list of 10 to the caller?

Reply:
Ahh I see. I didn't realize that was a requirement. Then yes, a map might be the only way.